### PR TITLE
IBX-719: BatchIterator

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-719
| Versions      | 3.3, master

Note: merge to 3.3 after 3.3.7 is out

### CODE REVIEW

File: docs/api/public_php_api_search.md [hunk 1]
```diff
@@ -87,6 +87,35 @@ As such, `query` is recommended when the search is based on user input.
 The difference between `query` and `filter` is only relevant when using Solr search engine.
 With the Legacy search engine both properties will give identical results.
 
+#### Batch searching
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:90]:
```suggestion
#### Batch Iterator
```

or 

```suggestion
#### Processing large result sets
```

File: docs/api/public_php_api_search.md [hunk 2]
```diff
@@ -87,6 +87,35 @@ As such, `query` is recommended when the search is based on user input.
...
+
+To search in a large result set, use `eZ\Publish\API\Repository\Iterator\BatchIterator`.
+`BatchIterator` divides the result set into smaller batches, which helps with sets that are too large to handle due to memory limits.
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:93]: `eZ\Publish\API\Repository\Iterator\BatchIterator` allows to iterate over all search or filtering results within memory constrains by automatically dividing the results into smaller batches (pages).

File: docs/api/public_php_api_search.md [hunk 3]
```diff
@@ -87,6 +87,35 @@ As such, `query` is recommended when the search is based on user input.
...
+
+`BatchIterator` takes one of the available adapters, and a [`Query` or `Filter` object](#search-with-query-and-filter).
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:95]:
```suggestion
`BatchIterator` takes one of the available adapters (`\eZ\Publish\API\Repository\Iterator\BatchIteratorAdapter` ) and optional batch size. For example: 
```

File: docs/api/public_php_api_search.md [hunk 4]
```diff
@@ -87,6 +87,35 @@ As such, `query` is recommended when the search is based on user input.
...
+$query = new LocationQuery;
+
+$iterator = new BatchIterator(new BatchIteratorAdapter\LocationSearchAdapter($this->searchService, $query));
+
+foreach ($iterator as $result) {
+    $output->writeln($result->valueObject->getContentInfo()->name);
+}
+```
+
+You can additionally define the batch size by setting `$iterator->setBatchSize()`.
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:107]:
```suggestion
You can also define the batch size by setting `$iterator->setBatchSize()`.
```

File: docs/api/public_php_api_search.md (hunk 2 above)
Comment (DominikaK) [reply docs/api/public_php_api_search.md:93]:
```suggestion
To search in a large result set, use `eZ\Publish\API\Repository\Iterator\BatchIterator`.
`BatchIterator` divides the results of search or filtering into smaller batches.
This enables iterating over results that are too large to handle due to memory constraints.
```

File: docs/api/public_php_api_search.md [hunk 5]
```diff
@@ -87,6 +87,36 @@ As such, `query` is recommended when the search is based on user input.
 The difference between `query` and `filter` is only relevant when using Solr search engine.
 With the Legacy search engine both properties will give identical results.
 
+#### Processing large result sets
+
+To search in a large result set, use `eZ\Publish\API\Repository\Iterator\BatchIterator`.
```
Comment (DominikaK) [inline docs/api/public_php_api_search.md:92]:
```suggestion
To process a large result set, use `eZ\Publish\API\Repository\Iterator\BatchIterator`.
```

**Review by adamwojs — APPROVED** [review]

**Pull request merged**